testcode/lib/clock-test.c

```diff
@@ -0,0 +1,100 @@
+/* Clock test program
+ *
+ * 25-Sep-2018, chris@groessler.org
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <errno.h>
+
+#ifdef __CC65__
+#include <conio.h>
+#include <cc65.h>
+
+static void exitfn(void)
+{
+    if (doesclrscrafterexit()) cgetc();
+}
+#endif /* #ifdef __CC65__ */
+
+static void print_time(void)
+{
+    struct tm *cur_tm;
+    time_t cur_time = time(NULL);
+    if (cur_time == -1) {
+        printf("time() failed: %s\n", strerror(errno));
+        return;
+    }
+    cur_tm = localtime(&cur_time);
+
+    printf("time: %s\n", asctime(cur_tm));
+    // DEBUG:
+    printf("mday=%d mon=%d year=%d\nhour=%d min=%d sec=%d\n", cur_tm->tm_mday, cur_tm->tm_mon, cur_tm->tm_year, cur_tm->tm_hour, cur_tm->tm_min, cur_tm->tm_sec);
+}
+
+int main(int argc, char **argv)
+{
+    static char c;
+    static int s;
+    static struct tm cur_time;
+    static struct timespec new_time;
+
+#ifdef __CC65__
+    atexit(exitfn);
```

Comment:
I sugesst to do this in the style of https://github.com/cc65/cc65/blob/master/libsrc/atari/targetutil/w2cas.c#L47-L49 and https://github.com/cc65/cc65/blob/master/samples/diodemo.c#L127-L130

Reply:
I can change that. I initially wrote it the way it is currently, and didn't change it so users can see that there are different ways to accomplish the same thing. Dou you want me to change it?

Reply:
To be honest I'd rather like to see us present a canonical way to do things.